recipes/default.rb

```diff
@@ -28,5 +28,5 @@
   product_name node['chef_client_updater']['product_name'] if node['chef_client_updater']['product_name']
   handle_zip_download_url node['chef_client_updater']['handle_zip_download_url'] if node['chef_client_updater']['handle_zip_download_url']
   event_log_service_restart node['chef_client_updater']['event_log_service_restart']
-  rubygems_url node['chef_client']['config']['rubygems_url'] if node['chef_client']['config']['rubygems_url']
+  rubygems_url node['chef_client_updater']['rubygems_url'] if node['chef_client_updater']['rubygems_url']
```

Comment:
We should probably just remove this line entirely and let folks that want to change the rubygems URL use the resource. The intention was that users would use the resource for anything outside the most basic usage.

Reply:
So rather than use the default recipe they'd just make their own wrapper cookbook and add the rubygems_url property?

Reply:
the "wrapper cookbook" concept here probably obscures more than it explains.

i usually use that to mean where you have a single-purpose wrapper around a base cookbook which are both attribute-driven and you are overriding things.

if you only use the resource out of this cookbook then that is the library cookbook pattern.  you're just using this cookbook as a library and using the resource out of it like you would use a resource out of core chef.

and in general you don't have to write a full blown acme_chef_client_updater wrapper cookbook which has a single recipe in it which calls the resource in this cookbook.  unless there is a whole lot more machinery to upgrading chef-client at the site which calls for it to be broken out into its own cookbook it may be a lot better off as a single recipe in a base cookbook (or even just a single resource in a larger recipe in the base cookbook).

Reply:
I think we're on the same page. I want to set some attributes in a policyfile and use the chef_client_updater::default recipe. I thought the intention was to remove the 
```
  rubygems_url node['chef_client_updater']['rubygems_url'] if node['chef_client_updater']['rubygems_url']
```
which would require calling the resource with it re-added if you need `rubygems_url. Including the above behavior is what I would expect.

Reply:
If we fully expect all functionality to be accomplished via the recipe then yeah we need to keep this. I'm a bigger fan of the resource, but it doesn't sound like everyone is on board with that.